File: tests/cpp/src/TestStorage.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include <cameo/api/cameo.h>

using namespace std;
using namespace cameo;

int main(int argc, char *argv[]) {

	This::init(argc, argv);

	string key = "eu.ill.cameo.test.testkey";
	string buffer = "{x:5}";

	try {
		string valueString = This::getCom().getKeyValue(key);
	}
	catch (const UndefinedKeyException& e) {
		cout << "Key is undefined: " << e.what() << endl;
	}

	This::getCom().storeKeyValue(key, buffer);

	try {
		This::getCom().storeKeyValue(key, buffer);
	}
	catch (const KeyAlreadyExistsException& e) {
		cout << "Key already exists: " << e.what() << endl;
	}

	try {
		string valueString = This::getCom().getKeyValue(key);
		cout << "Value is " << valueString << endl;

		This::getCom().removeKey(key);
	}
	catch (const UndefinedKeyException& e) {
	}

	try {
		This::getCom().getKeyValue(key);
	}
	catch (const UndefinedKeyException& e) {
		cout << "Cannot remove key : " << e.what() << endl;
	}

	cout << "Finished the application" << endl;

	This::terminate();

	return 0;
}

File: tests/java/src/eu/ill/cameo/test/TestStorage.java
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

package eu.ill.cameo.test;

import org.json.simple.JSONObject;
import org.json.simple.parser.ParseException;

import eu.ill.cameo.api.base.KeyAlreadyExistsException;
import eu.ill.cameo.api.base.This;
import eu.ill.cameo.api.base.UndefinedKeyException;
import eu.ill.cameo.common.messages.JSON;


public class TestStorage {

	public static void main(String[] args) {
		
		This.init(args);

		String key = "eu.ill.cameo.test.testkey";
		
		JSONObject valueObject = new JSONObject();
		valueObject.put("x", 12);
		
		try {
			This.getCom().getKeyValue(key);
		}
		catch (UndefinedKeyException e) {
			System.out.println("Key is undefined: " + e.getMessage());
		}
		
		try {
			This.getCom().storeKeyValue(key, valueObject.toJSONString());
		}
		catch (KeyAlreadyExistsException e) {
		}
		
		try {
			This.getCom().storeKeyValue(key, valueObject.toJSONString());
		}
		catch (KeyAlreadyExistsException e) {
			System.out.println("Key already exists: " + e.getMessage());
		}
		
		try {
			String valueString = This.getCom().getKeyValue(key);
			valueObject = JSON.parse(valueString);
			
			System.out.println("x = " + JSON.getInt(valueObject, "x"));
			
			This.getCom().removeKey(key);
		}
		catch (UndefinedKeyException e) {
		}
		catch (ParseException e) {
		}
		
		try {
			This.getCom().getKeyValue(key);
		}
		catch (UndefinedKeyException e) {
			System.out.println("Cannot remove key: " + e.getMessage());
		}

		System.out.println("Finished the application");
		
		This.terminate();
	}

}